.github/workflows/agw-workflow.yml

```diff
@@ -0,0 +1,311 @@
+        run: |
+            docker run -v ${{ env.MAGMA_ROOT }}:/magma -v ${{ env.MAGMA_ROOT }}/lte/gateway/configs:/etc/magma -i magma/c_cpp_build:latest /bin/bash -c "cd /magma/lte/gateway;make test_sctpd BUILD_TYPE=RelWithDebInfo;"
+      - name: Run mme tests with Debug build type
+        run: |
+            docker run -v ${{ env.MAGMA_ROOT }}:/magma -v ${{ env.MAGMA_ROOT }}/lte/gateway/configs:/etc/magma -i magma/c_cpp_build:latest /bin/bash -c "cd /magma/lte/gateway;make test_oai BUILD_TYPE=Debug;"
+      - name: Run mme tests with RelWithDebInfo build type
+        run: |
+            docker run -v ${{ env.MAGMA_ROOT }}:/magma -v ${{ env.MAGMA_ROOT }}/lte/gateway/configs:/etc/magma -i magma/c_cpp_build:latest /bin/bash -c "cd /magma/lte/gateway;make test_oai BUILD_TYPE=RelWithDebInfo;"
+      # yamllint enable
+      # Notify ci channel when failing
+      # Plugin info: https://github.com/marketplace/actions/slack-notify
+      - name: Notify failure to slack
+        if: failure() && github.ref == 'refs/heads/master'
+        uses: rtCamp/action-slack-notify@v2.0.2
+        env:
+          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK_CI }}
+          SLACK_TITLE: "Github action mme_test failed"
+          SLACK_USERNAME: "AGW workflow"
+          SLACK_ICON_EMOJI: ":boom:"
+          SLACK_COLOR: "#FF0000"
+  mme-clang-tidy:
+    runs-on: ubuntu-latest
+    env:
+      MAGMA_ROOT: "${{ github.workspace }}"
+      BRANCH: "${{ github.base_ref }}"
+      REVISION: "${{ github.sha }}"
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run mme-clang-tidy
+        uses: nick-invision/retry@v2
+        # yamllint disable rule:line-length
+        with:
+          command: |
+            cd ${{ env.MAGMA_ROOT }}/lte/gateway/docker/mme
+            docker build --tag magma-mme-build --file Dockerfile.ubuntu20.04 ../../../../
+            docker run --env BRANCH=${{ env.BRANCH}} --env REVISION=${{ env.REVISION }} --volume ${{ env.MAGMA_ROOT }}:/magma --interactive magma-mme-build:latest /bin/bash --cpu-shares 'cd /magma/lte/gateway;make clang_tidy_oai_upload'
+          timeout_minutes: 20
+      # yamllint enable
+      # Notify ci channel when failing
+      # Plugin info: https://github.com/marketplace/actions/slack-notify
+      - name: Notify failure to slack
+        if: failure() && github.ref == 'refs/heads/master'
+        uses: rtCamp/action-slack-notify@v2.0.2
+        env:
+          SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK_CI }}
+          SLACK_TITLE: "Github action mme-clang-tidy failed"
+          SLACK_USERNAME: "AGW workflow"
+          SLACK_ICON_EMOJI: ":boom:"
+          SLACK_COLOR: "#FF0000"
+  mme-clang-warnings:
+    runs-on: ubuntu-latest
+    env:
+      MAGMA_ROOT: "${{ github.workspace }}"
+      BRANCH: "${{ github.base_ref }}"
+      REVISION: "${{ github.sha }}"
+    steps:
+      - uses: actions/checkout@v2
+      - name: Run mme-clang-warnings
+        uses: nick-invision/retry@v2
```

Comment:
Avoid using this module, it does not work well...

Reply:
sure!.. shall i use the manual retry? If so, how many retries?

Reply:
I think for this specific step, you don't need to retry. There was no manual retry in circle ci

Reply:
I noticed I am using the retry module a lot :/
Shall I do just one single run? or a manual retry like we did in the previous workflow?

Reply:
just don't use it, but some circle ci step had a specific timeout.
For this override the timeout at the step level. refer here https://stackoverflow.com/questions/59073731/set-default-timeout-on-github-action-pipeline

Reply:
sure.. let me update those.